internal/utils/endorsement.go

```diff
@@ -3,40 +3,89 @@ package utils
 import (
 	"fmt"
 
-	"github.com/hyperledger/fabric-protos-go/peer/lifecycle"
-	"github.com/hyperledger/fabric/bccsp/factory"
-	"github.com/hyperledger/fabric/common/cauthdsl"
-	"github.com/hyperledger/fabric/msp"
-	"github.com/hyperledger/fabric/msp/mgmt"
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	pmsp "github.com/hyperledger/fabric-protos-go/msp"
+	pb "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/pkg/errors"
 )
 
-func NewPolicyEvaluator() *PolicyEvaluator {
-	manager := mgmt.NewDeserializersManager(factory.GetDefault())
-	return &PolicyEvaluator{
-		IdentityDeserializer: manager.GetLocalDeserializer(),
-	}
+type PolicyEvaluatorInterface interface {
+	EvaluateIdentity(policyBytes []byte, identityBytes []byte) error
 }
 
 type PolicyEvaluator struct {
-	msp.IdentityDeserializer
 }
 
+func NewPolicyEvaluator() PolicyEvaluatorInterface {
+	return &PolicyEvaluator{}
+}
+
+// EvaluateIdentity checks that the provided identity is a valid endorser as specified in the endorsement policy
```

Comment:
My understanding is that this part is to check whether the enclave is a valid endorser.

That said, do we need this whole part at all?

We definitely _need_ that in an FPC/Fabric integration where the enclave acts as a _Fabric_-endorser and signs the response.
However, in the new architecture, this is not the case anymore. The enclave acts as an _FPC_-endorser and its signature is checked by the FPC (_not Fabric_) validation chaincode. Most importantly, FPC uses ERCC to control which enclaves can FPC-endorse a transaction and, therefore, pass the validation check. Hence, it is not necessary to check whether the Fabric validation logic can recognize an enclave's signature through MSP.

Reply:
yes, all correct. We can keep this code for FULL FPC ercc implementation. Note that the policy evaluated got invalided with c4a0ace during review session with Michael :D